src/els_sup.erl

```diff
@@ -40,6 +40,8 @@ init([]) ->
               , period    => 60
               },
   {ok, Transport} = application:get_env(erlang_ls, transport),
+  %% Restrict access to stdio when using that transport
+  (Transport =/= els_stdio) andalso restrict_stdio_access(),
```

Comment:
Isn't this the opposite of what you want? You call `restrict_stdio_access()` for every transport _except for_ `els_stdio`.

Reply:
It looks like @dszoboszlay is correct. I also suggest we do not rely on the `andalso` semantics for this, since it makes the code less readable. I would rather:

```
restrict_stdio_access(els_stdio) ->
  restrict_stdio_access();
restrict_stdio_access(_) ->
  ok.
```

Reply:
🤦

Reply:
#532 532